### PR TITLE
add io.github.xiayifang.goldendict_ng

### PR DESCRIPTION
**Pull request by shenlebantongying**
Add this package https://github.com/xiaoyifang/goldendict-ng

Question: About the domain requirement, should we change it to `com.github.goldendict_ng` or `io.github.xiayifang.goldendict_ng` ?

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by shenlebantongying** [comment]
bot, build org.xiaoyifang.GoldenDict_NG

**Comment by shenlebantongying** [comment]
bot, build io.github.xiayifang.goldendict_ng

**Comment by shenlebantongying** [comment]
bot, build io.github.xiayifang.goldendict_ng

**Comment by shenlebantongying** [comment]
bot, build io.github.xiayifang.goldendict_ng

**Comment by shenlebantongying** [comment]
@JakobDev Thank you for reviewing :)

The id & metadata.xml is changed to `io.github.xiayifang.goldendict_ng`. Related changes will be merged to upstream.

The git source repo right now points to my repo, which will be changed to the real upstream once all requirements are fulfilled. I think it should be OK to change that url later.

**Comment by shenlebantongying** [comment]
bot, build io.github.xiayifang.goldendict_ng

**Comment by shenlebantongying** [comment]
bot, build io.github.xiayifang.goldendict_ng

**Comment by shenlebantongying** [comment]
bot, build io.github.xiayifang.goldendict_ng

**Comment by shenlebantongying** [comment]
bot, build io.github.xiayifang.goldendict_ng

**Comment by shenlebantongying** [comment]
This is ready :)

**Comment by barthalion** [comment]
```
~ 5.6s ❱ flatpak run io.github.xiayifang.goldendict_ng
bwrap: Can't mkdir /app/lib/ffmpeg: Read-only file system
error: ldconfig failed, exit status 256
```

You need to create this directory as part of the build.

**Comment by shenlebantongying** [comment]
bot, build io.github.xiayifang.goldendict_ng

**Comment by shenlebantongying** [comment]
bot, build io.github.xiayifang.goldendict_ng

**Comment by shenlebantongying** [comment]
> bwrap: Can't mkdir /app/lib/ffmpeg: Read-only file system
> You need to create this directory as part of the build.

I added the path as others do. However, I cannot reproduce the problem on my system. Do I need an immutable OS like feodra sliver blue to reproduce?

Thank you for review @barthalion :)

File: io.github.xiayifang.goldendict_ng.json [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      ]
+    },
+    {
+      "name": "goldendict-ng",
+      "buildsystem": "cmake-ninja",
+      "build-options": {
+        "env": {
+          "LD_LIBRARY_PATH": "${LD_LIBRARY_PATH}:/app/lib/:/usr/lib"
+        }
+      },
```
Comment (hfiguiere) [inline io.github.xiayifang.goldendict_ng.json:156]: this shouldn't be necessary.

File: io.github.xiayifang.goldendict_ng.json [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      ],
+      "build-commands": [
+        "install -Dm644 redist/icons/goldendict.png ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png",
+        "desktop-file-edit --set-key=\"Icon\" --set-value=\"${FLATPAK_ID}\" redist/${FLATPAK_ID}.desktop"
+      ],
+      "cleanup": [
+        "/share/pixmaps"
+      ],
+      "cleanup-commands": [
+        "mkdir -p /app/lib/ffmpeg"
```
Comment (hfiguiere) [inline io.github.xiayifang.goldendict_ng.json:171]: this should be in the `build-commands`.

File: io.github.xiayifang.goldendict_ng.json (hunk 1 above)
Comment (shenlebantongying) [reply io.github.xiayifang.goldendict_ng.json]: fixd

File: io.github.xiayifang.goldendict_ng.json (hunk 2 above)
Comment (shenlebantongying) [reply io.github.xiayifang.goldendict_ng.json]: Movded to `build-commands`

I copied this from others :sweat_smile:

https://github.com/search?q=org%3Aflathub+++++++cleanup-commands+mkdir+-p+%2Fapp%2Flib%2Fffmpeg&type=code

**Comment by barthalion** [comment]
The app ID states `xiayifang`. If that's not the correct ID, you need to open a new PR with the corrected ID, then mark the wrong one as EOL: https://docs.flathub.org/docs/for-app-authors/maintanance#end-of-life

**Comment by shenlebantongying** [comment]
🤦🤦🤦🤦🤦🤦 My bad

**Pull request closed without merging**